### PR TITLE
Run codesign on osx-arm64

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Review by beckermr — COMMENTED** [review]
Does this pr sign all binaries or only those which have a prefixed replaced?

**Comment by isuruf** [comment]
Only those which have a prefixed replaced which is all binaries produced by the conda compilers.

**Comment by beckermr** [comment]
Ok. So we won't be able to sign things if we are doing binary repackaging?

**Comment by beckermr** [comment]
Or I should say, installing a thing from binary repackaging.

**Comment by isuruf** [comment]
Nope. That should be done in `conda-build` if `binary_has_prefix_files` is not turned off. (Usually binary repackages turn it off)

**Comment by beckermr** [comment]
Ok. So we do need the pr on conda build.

**Comment by beckermr** [comment]
What about statically linked binaries?

**Comment by isuruf** [comment]
Those have rpath in them if they were compiled using conda compilers, so they'll get replaced as well.

**Comment by erykoff** [comment]
Latest changes check if a file has been _updated_ on whether it should do the signing.  But we don't have any guarantee that it was signed in the first place, right?  I think we either need to check if it has a valid signature (`codesign -v file`) or just sign everything.

**Comment by isuruf** [comment]
If a package was not updated here, that means there was something wrong with the metadata in the conda package. That code path should not be triggered at all, but I added it just in case.

**Comment by isuruf** [comment]
ping on this

**Review by chenghlee — APPROVED** [review]

File: conda/core/portability.py [hunk 1]
```diff
@@ -56,7 +60,11 @@ def _update_prefix(original_data):
 
         return data
 
-    update_file_in_place_as_binary(realpath(path), _update_prefix)
+    updated = update_file_in_place_as_binary(realpath(path), _update_prefix)
+
+    if updated and mode == FileMode.binary and subdir == "osx-arm64" and sys.platform == "darwin":
+        # Apple arm64 needs signed executables
+        subprocess.run(['codesign', '-s', '-', '-f', realpath(path)])
```
[the remark this answers is not in the document]
Comment (chenghlee) [reply conda/core/portability.py]: Fix in commit d2f2628

**Comment by beckermr** [comment]
huzzah!

**Comment by erykoff** [comment]
Fantastic!

**Pull request merged**